### PR TITLE
add data augmentation scripts for working with the green box

### PR DESCRIPTION
**Pull request by minhnh**
* resolves minhnh/dataset_interface#1
* resolves minhnh/dataset_interface#6
* handles user interactions for each augmentation scripts
* fix out-of-bound issues with previous transformation matrices
* clean up masks and projections of segmented objects
* add documentation for each augmentation components

### CODE REVIEW

File: augmentation/data_augmenter.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
```
Comment (alex-mitrevski) [inline augmentation/data_augmenter.py]: Is there a need for two augmentation scripts (`artificial_images_generator.py` and this one)? Just by skimming the code, the scripts are pretty much the same with minor differences.

Comment (minhnh) [reply augmentation/data_augmenter.py]: I haven't refactored there yet. Noted that they're pretty much the same.

**Comment by minhnh** [comment]
Training with `tensorflow/models` scripts will resolve b-it-bots/mas_perception#34

**Comment by minhnh** [comment]
@alex-mitrevski and @ErickKramer please take a look and review, so I can merge this PR.

**Comment by alex-mitrevski** [comment]
I'll try to take a look later today.

File: docs/image_augmentation.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
+# Image augmentation
+
+TODO(minhnh) add and link to Jupyter notebook example
```
Comment (alex-mitrevski) [inline docs/image_augmentation.md]: Can you remove this comment?

File: docs/image_augmentation.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+* [Background segmentation](#background-segmentation)
+* [Generating images and bounding box annotations from segmented masks](#generating-images-and-bounding-box-annotations-from-segmented-masks)
+* [Colect images from a ROS topic](#colect-images-from-a-ros-topic)
```
Comment (alex-mitrevski) [inline docs/image_augmentation.md]: Typo: colect -> collect

**Pull request merged**